### PR TITLE
feat: add page titles using React Helmet

### LINKED ISSUE #1294: Show Page Name Instead of Default Heading
### Description:

In the current setup of our app, the page title doesn't change by default when navigating to different pages. To ensure proper SEO and provide a clear indication of the page's content, we should consider showing the page name in the title instead of relying on a default heading.

image

### Steps to Reproduce:

1. Navigate to any page within the application.
2. Observe the browser tab title – it remains static and doesn't reflect the current page's content.

### Expected Behavior:

The browser tab title should dynamically update to show the name of the current page, providing better context for users and search engines.

### PR DESCRIPTION
**Pull request by mustajab-ikram**
I have completed the task related to Issue #1294. I have added the Page Title functionality using React Helmet Async lib.
Please take a look at the PR when you get a chance @charlesBochet. Let me know if there are any further changes or improvements you would like to see.
image
image
image
image
image

closes #1294

### CODE REVIEW

**Comment by charlesBochet** [comment]
Looks great @mustajab-ikram! I think you need to fix the tests (I suspect, you need to also add the HelmetProvider to our Pages Decorators?)

To run them:
`yarn storybook:dev`

and in another terminal
`yarn storybook:coverage`

Once they are green, we can merge!

File: front/src/utils/title-utils.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    case AppPath.PeoplePage:
+      return 'People';
+    case AppPath.CompaniesPage:
+      return 'Companies';
+    case AppPath.TasksPage:
+      return 'Tasks';
+    case AppPath.OpportunitiesPage:
+      return 'Opportunities';
+    case `${AppBasePath.Settings}/${SettingsPath.ProfilePage}`:
+      return 'Profile Settings';
```
Comment (charlesBochet) [inline front/src/utils/title-utils.ts:28]: Profile

File: front/src/utils/title-utils.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    case `${AppBasePath.Settings}/${SettingsPath.Experience}`:
+      return 'Experience';
+    case `${AppBasePath.Settings}/${SettingsPath.WorkspaceMembersPage}`:
+      return 'Workspace Members Settings';
+    case `${AppBasePath.Settings}/${SettingsPath.Workspace}`:
+      return 'Workspace Settings';
```
Comment (charlesBochet) [inline front/src/utils/title-utils.ts:34]: Workspace

File: front/src/utils/title-utils.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    case AppPath.TasksPage:
+      return 'Tasks';
+    case AppPath.OpportunitiesPage:
+      return 'Opportunities';
+    case `${AppBasePath.Settings}/${SettingsPath.ProfilePage}`:
+      return 'Profile Settings';
+    case `${AppBasePath.Settings}/${SettingsPath.Experience}`:
+      return 'Experience';
+    case `${AppBasePath.Settings}/${SettingsPath.WorkspaceMembersPage}`:
+      return 'Workspace Members Settings';
```
Comment (charlesBochet) [inline front/src/utils/title-utils.ts:32]: Workspace Members

File: front/src/pages/people/PersonShow.tsx [hunk 4]
```diff
@@ -63,66 +64,75 @@ export function PersonShow() {
...
-                      </EditableFieldDefinitionContext.Provider>
-                    );
-                  })}
-                </EditableFieldEntityIdContext.Provider>
-              </EditableFieldMutationContext.Provider>
-            </PropertyBox>
-          </ShowPageLeftContainer>
-          <ShowPageRightContainer
+    <>
+      <Helmet>
```
Comment (charlesBochet) [inline front/src/pages/people/PersonShow.tsx:68]: Maybe extract this to a PageTitle component under @/ui/utilities/page-title

File: front/src/utils/title-utils.ts (hunk 2 above)
Comment (mustajab-ikram) [reply front/src/utils/title-utils.ts]: done

File: front/src/utils/title-utils.ts (hunk 3 above)
Comment (mustajab-ikram) [reply front/src/utils/title-utils.ts]: done

File: front/src/pages/people/PersonShow.tsx (hunk 4 above)
Comment (mustajab-ikram) [reply front/src/pages/people/PersonShow.tsx]: created a separate component for this

**Comment by charlesBochet** [comment]
@mustajab-ikram Looks good, we still need to fix those tests :)

**Comment by mustajab-ikram** [comment]
> @mustajab-ikram Looks good, we still need to fix those tests :)

done

**Review by charlesBochet — APPROVED** [review]
Thank you, I like the dev experience

**Pull request merged**